src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.NativeCrypto.cs

```diff
@@ -51,6 +51,33 @@ internal static partial class NativeCrypto
         [DllImport(Libraries.CryptoInterop)]
         private static extern int GetAsn1StringBytes(IntPtr asn1, byte[] buf, int cBuf);
 
+        [DllImport(Libraries.CryptoInterop)]
+        internal static extern int GetX509StackFieldCount(SafeX509StackHandle stack);
+
+        /// <summary>
+        /// Gets a pointer to a certificate within a STACK_OF(X509). This pointer will later
+        /// be freed, so it should be cloned via new X509Certificate2(IntPtr)
+        /// </summary>
```

Comment:
The fact that the pointer has a short lifespan, and needs to be duplicated.  Most everything else here returns things whose lifetime is caller-determined (and thus marshalls through as a SafeHandle).